srv/modules/runners/replace.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=too-few-public-methods,modernize-parse-error
+"""
+Runner to remove a single osd
+"""
+
+from __future__ import absolute_import
+from __future__ import print_function
+import time
+import logging
+import os
+# pylint: disable=import-error,3rd-party-module-not-gated,redefined-builtin
+import salt.client
+import salt.runner
+
+log = logging.getLogger(__name__)
+
+
+def help_():
+    """
+    Usage
+    """
+    usage = ('salt-run replace.osd id [id ...][force=True]:\n\n'
+             '    Removes an OSD from a minion\n'
+             '\n\n')
+    print(usage)
+    return ""
+
+
+def osd(*args, **kwargs):
+    """
+    Remove an OSD gracefully or forcefully on the minion
+
+    Note: If I were not in Salt, this whole routine would become a library.
+    This runner can be called by remove.osd and is only different by three
+    commands.  However, we have one runner call and multiple module calls below.
+    Trying to refactor these module calls into a __utils__ would require yet
+    another module.  Debugging all of this at runtime would be add to the fun,
+    so I didn't.
+    """
+    # Parameters for osd.remove module
+    supported = ['force', 'timeout', 'delay']
+    passed = ["{}={}".format(k, v) for k, v in kwargs.items() if k in supported]
```

Comment:
Would a notification that someone passed _unsupported_ parameters be helpful here? That'd also help people that did a typo.

Reply:
The underscore variables get passed from Salt when called via a state file, iirc.  The warning would be a wall of text without extra processing trying to separate Salt internal variables.  I thought the debug would be enough.